frontend/src/components/AddClassModal.tsx

```diff
@@ -138,6 +158,33 @@ export class AddClassModal extends React.Component<
     }
   }
 
+  isCourseInSchedule(courseToAdd: ScheduleCourse, schedule: DNDSchedule) {
```

Comment:
I would add java doc style comments explaining what each of the functions that you wrote do